### PR TITLE
Auto enable dependencies

### PR DESCRIPTION
**Pull request by sfmskywalker**
This ensures that any dependency feature currently not enabled will automatically be enabled before the shell is activated, effectively preventing issues as described in #3132.

### CODE REVIEW

File: src/Orchard/Environment/ShellBuilders/CompositionStrategy.cs [hunk 1]
```diff
@@ -64,16 +57,41 @@ public class CompositionStrategy : ICompositionStrategy {
...
+            if (autoEnabledDependencyFeatures.Any()) {
+                // Add any dependencies previously not enabled to the shell descriptor.
+                descriptor.Features = descriptor.Features.Concat(autoEnabledDependencyFeatures.Select(x => new ShellFeature { Name = x })).ToList();
+                Logger.Information("Automatically enabled the following dependency features: {0}.", String.Join(", ", autoEnabledDependencyFeatures));
+            }
+
             return result;
         }
 
+        private IEnumerable<string> ExpandDependencies(IDictionary<string, FeatureDescriptor> availableFeatures, IEnumerable<string> features) {
```
Comment (Piedone) [inline src/Orchard/Environment/ShellBuilders/CompositionStrategy.cs]: This would be better as a helper common with FeatureManager.

**Comment by sebastienros** [comment]
Please fix the conflicts

**Comment by sfmskywalker** [comment]
Will do.

**Comment by sfmskywalker** [comment]
Fixed

**Comment by jeremygrajales** [comment]
I get build errors on dev that seem to result from this feature because of the deletion of several .js files still referenced in Orchard.Azure.MediaServices.csproj
Should this be an issue?

**Comment by sfmskywalker** [comment]
I'll check.

**Comment by sfmskywalker** [comment]
Just pushed the missing files.

**Comment by jeremygrajales** [comment]
Thanks!

**Pull request merged**